src/ctia/http/routes/bulk.clj

```diff
@@ -57,27 +68,47 @@
                    :sighting       read-sighting
                    :ttp            read-ttp) %))
 
+(defn with-long-id-fn
+  "return the with-long-id function provided an entity type key"
+  [k]
+  (case k
+    :actor          act-ent/with-long-id
+    :campaign       cam-ent/with-long-id
+    :coa            coa-ent/with-long-id
+    :exploit-target ept-ent/with-long-id
+    :feedback       fbk-ent/with-long-id
+    :incident       inc-ent/with-long-id
+    :indicator      ind-ent/with-long-id
+    :judgement      jud-ent/with-long-id
+    :sighting       sig-ent/with-long-id
+    :ttp            ttp-ent/with-long-id))
+
 (defn create-entities
   "Create many entities provided their type and returns a list of ids"
   [entities entity-type login]
-  (->> entities
-       (map #(try (flows/create-flow
-                   :entity-type entity-type
-                   :realize-fn (realize entity-type)
-                   :store-fn (create-fn entity-type)
-                   :identity login
-                   :entity %)
-                  (catch Exception e
-                    (do (log/error (pr-str e))
-                        nil))))
-       (map :id)))
+  (let [with-long-id (with-long-id-fn entity-type)]
+    (->> entities
+         (map #(try
```

Comment:
This function that starts with `try` is quite large, and breaks up the threading macro. I suggest labeling it in the `let` block and calling map on that function.